### DISCUSSION TITLE
smart search and router dont like each other

### DISCUSSION THREAD

**Discussion by nzyeehaa** [joomla/joomla-cms#38108] [General] [answered]
I have a menu item setup as **search** so search can be accessed by menu or by module.
and therefore all searches should show with a sef url https://website.com/search/...
`modules>smart search > set itemID = search`

when you search for an item, the url is https://website.com/search?q=motor

and click the link on a search result -  its url and itemid is shown as the home page - $itemid=101 = homepage
`<?php echo $itemid; ?> `101 
https://website.com/?view=article&id=600:title&catid=100
- not a good move

this means if I have 600 items, with smart search enabled - I also have 600+ home pages

~at bare minimum it should be ...https://website.com/search?view=article&id=600:title&catid=100
and $itemid =109= search~

note to self - remember to add
search = noindex nofollow

**Comment by brianteeman** [comment]
I cannot replicate this.

With the sample data installed I search for "module" using the module on the home page

The results are displayed on /search?q=module

I click on the result and the page that is loaded is correctly
/blog/your-modules

If I turn off sef urls so I can check the itemid and repeat the same test

The results are displayed on /index.php?q=module&option=com_finder&view=search&Itemid=115

I click on the result and the page that is loaded is correctly
index.php?option=com_content&view=article&id=5:your-modules&catid=8&Itemid=102

**Comment by nzyeehaa** [comment]
@brianteeman thanks for taking the time to check it out
your results worked as there was a menu item /blog/your-modules

heres some results from a clean install (but only J 4.1.2)

clean install 
no sample data
.htaccess
config > SEF urls = yes
config > use URL rewriting = yes
added...
>menu = "**finder** = smartsearch"
>menu = "two" = single article
>modules > smart search > set itemID = "finder" & opensearch off
>categories > cattwo
>content > add 1 article (uncategorised and feature on home page)
>content > added articles  with "test" words included - put into cattwo
>content > one article called "two", also in cattwo, link to menu "two" 

module > search "test"

1st result `http://j41.ld/?view=article&id=1:test&catid=2`  _**(is ok - could be better - page is uncategorised and featured on home page)**_
+ breadcrumbs = You are here:  `[Home](http://j41.ld/) [Uncategorised](http://j41.ld/?view=category&id=2) test` 

2nd result `http://j41.ld/two`  _**(looks good - page two, linked to menu two = _SEF works with articles listed in a menu_ )**_
+ breadcrumbs = You are here:  `[Home](http://j41.ld/)two` _**(looks good)**_

3rd result `http://j41.ld/?view=article&id=8:published-art&catid=8` _**(not so good - page is not featured on home page )
+ breadcrumbs = `[Home](http://j41.ld/) [cattwo](http://j41.ld/?view=category&id=8)published art `
[ should be` finder?view=category&id=8` or `finder?cattwo&published art` ]


> ** extend the test **
> menu > category list "list cattwo"
> 3rd result now shows `http://j41.ld/list-cattwo/published-art`
> + breadcrumbs = `[Home](http://j41.ld/) [list cattwo](http://j41.ld/list-cattwo) published art
`

**Comment by brianteeman** [comment]
>I have a menu item setup as search so search can be accessed by menu or by module.
and therefore all searches should show with a sef url https://website.com/search/...
modules>smart search > set itemID = search

That is for the itemid to be used for the search results page. NOT for the links to the items in the search results.

Joomla will always use the itemid of the article. This could be a direct menu link or a category menu link etc.

If there is no suitable itemid then joomla will always use the itemid of the home item

**Comment by nzyeehaa** [comment, marked as answer]
in result one above...
breadcrumbs show `[Home][Uncategorised]test`
search URL shows` ?view=article&id=1:test&catid=2`
neither of these would be correct for SEO as the article is on the home page so should show base domain/ and page content is accessible by different URL's

in result three above...
breadcrumbs show `[Home][cattwo]published art`
search URL shows `?view=article&id=8:published-art&catid=8`

in extended test above...
menu URL` /list-cattwo/published-art`
breadcrumb URL `[Home] [list cattwo] published art`
search URL  `/list-cattwo/published-art`

**Moral of that story,** if you need to improve SEO in Joomla and reduce duplicate content...
`**The menu is what drives your SEF URL's**`
setup a new "spare menu" with links to each category, then to each category/subcategory to replicate your category tree
dont add a module for the spare menu (it is only used for SEO)

Now the breadcrumbs, search URL and links will all line up as per your category structure
If you list a category on your main menu, you may need to remove it from your spare menu first

PS: it doesn't fix the home page featured problem, maybe a _redirect_ from `uncategorised` to `home/`? - it will depend on how you organise your site

**Reply by nzyeehaa** [reply]
also seems to be other router issues happening here #38118

**Comment by Hackwar** [comment]
This is a bug in your sites configuration, not from Joomla. For support questions like this, please go to forum.joomla.org
